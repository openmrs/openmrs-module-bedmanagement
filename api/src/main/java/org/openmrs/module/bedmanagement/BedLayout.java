package org.openmrs.module.bedmanagement;

import org.openmrs.Patient;
import org.openmrs.module.bedmanagement.entity.BedTagMap;
import org.openmrs.module.bedmanagement.entity.BedType;

import java.util.Set;

public class BedLayout {
    private Integer rowNumber;
    private Integer columnNumber;
    private String bedNumber;
    private Integer bedId;
    private String bedUuid;
    private String status;
    private BedType bedType;
    private String location;
    private Patient patient;
    private Set<BedTagMap> bedTagMaps;

    public String getBedNumber() {
        return bedNumber;
    }

    public void setBedNumber(String bedNumber) {
        this.bedNumber = bedNumber;
    }

    public Integer getBedId() {
        return bedId;
    }

    public void setBedId(Integer bedId) {
        this.bedId = bedId;
    }

    public String getBedUuid() {
        return bedUuid;
    }

    public void setBedUuid(String bedUuid) {
        this.bedUuid = bedUuid;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Integer getRowNumber() {
        return rowNumber;
    }

    public void setRowNumber(Integer rowNumber) {
        this.rowNumber = rowNumber;
    }

    public Integer getColumnNumber() {
        return columnNumber;
    }

    public void setColumnNumber(Integer columnNumber) {
        this.columnNumber = columnNumber;
    }

    public BedType getBedType() {
        return bedType;
    }

    public void setBedType(BedType bedType) {
        this.bedType = bedType;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public Patient getPatient() {
        return patient;
    }

    public void setPatient(Patient patient) {
        this.patient = patient;
    }

    @Override
    public String toString() {
        return "BedLayout{" +
                "rowNumber=" + rowNumber +
                ", columnNumber=" + columnNumber +
                ", bedNumber='" + bedNumber + '\'' +
                ", bedId=" + bedId +
                ", bedUuid=" + bedUuid +
                ", status='" + status + '\'' +
                ", bedType=" + bedType +
                ", location='" + location + '\'' +
                '}';
    }

    public Set<BedTagMap> getBedTagMaps() {
        return bedTagMaps;
    }

    public void setBedTagMaps(Set<BedTagMap> bedTagMaps) {
        this.bedTagMaps = bedTagMaps;
    }
}
