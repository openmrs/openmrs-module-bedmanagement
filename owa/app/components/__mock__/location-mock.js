Object.defineProperties(window.location, {
    host : {
        writable: true,
        value: '192.168.33.10'
    },
    hostname : {
        writable: true,
        value: '192.168.33.10'
    },
    href :{
        writable: true,
        value: 'https://192.168.33.10/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    origin :{
        writable: true,
        value: 'https://192.168.33.10'
    },
    pathname :{
        writable: true,
        value: '/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    port :{
        writable: true,
        value: ''
    },
    protocol :{
        writable: true,
        value: 'https:'
    }
});