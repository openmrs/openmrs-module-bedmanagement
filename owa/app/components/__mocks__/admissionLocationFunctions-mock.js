import pickBy from 'lodash.pickby';
import {admissionLocations, visitLocations, bedTypes} from 'utilities/__tests__/testData.json';

const admissionLocationFunctionsMock = {
    setActiveLocationUuid: (admissionLocationUuid) => jest.fn(),
    getActiveLocationUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    setState: (data) => jest.fn(),
    getState: () => jest.fn(),
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    getVisitLocations: () => {
        return visitLocations;
    },
    getBedTypes: () => {
        return bedTypes;
    },
    getAdmissionLocationByUuid: (admissionLocationUuid) => {
        return typeof admissionLocations[admissionLocationUuid] !== 'undefined' ?
            admissionLocations[admissionLocationUuid] : null;
    },
    getParentAdmissionLocation: (admissionLocationUuid) => {
        const parentAdmissionLocationUuid = admissionLocations[admissionLocationUuid].parentAdmissionLocationUuid;
        return parentAdmissionLocationUuid != null ? admissionLocations[parentAdmissionLocationUuid] : null;
    },
    getChildAdmissionLocations : (admissionLocationUuid) => {
        return  pickBy(admissionLocations, (admissionLocation) => {
            return admissionLocation.parentAdmissionLocationUuid == admissionLocationUuid;
        });
    },
    reFetchAllAdmissionLocations : () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default admissionLocationFunctionsMock;
