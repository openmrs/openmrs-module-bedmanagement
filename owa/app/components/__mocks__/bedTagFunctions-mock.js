import _ from 'lodash';

import {bedTags} from 'utilities/__tests__/testData.json';

const bedTagFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTags: () => {
        return bedTags;
    },
    getBedTagByUuid: (bedTagUuid) => {
        return _.find(bedTags, function (bedTag) {
            return bedTag.uuid == bedTagUuid;
        });
    },
    fetchBedTags: () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default bedTagFunctionsMock;
