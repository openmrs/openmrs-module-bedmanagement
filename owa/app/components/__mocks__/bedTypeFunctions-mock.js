import _ from 'lodash';

import {bedTypes} from 'utilities/__tests__/testData.json';

const bedTypeFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTypes: () => {
        return bedTypes;
    },
    getBedTypeById: (bedTypeId) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.id == bedTypeId;
        });
    },
    getBedTypeName: (bedTypeName) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.name == bedTypeName;
        });
    },
    fetchBedTypes: () => {
        return bedTypes;
    },
    notify : (notifyType, message) => jest.fn()
};

export default bedTypeFunctionsMock;
