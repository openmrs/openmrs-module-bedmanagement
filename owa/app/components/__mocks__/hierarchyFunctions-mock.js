import {admissionLocations} from 'utilities/__tests__/testData.json';

const hierarchyFunctionMock = {
    getActiveUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    setActiveUuid: (locationUuid) => jest.fn(),
    setState: (stateData) => jest.fn(),
    toggleIsOpen: (locationUuid, isOpen) => jest.fn(),
    setAdmissionLocationIsOpen: (locationUuid, isOpen) => jest.fn()
};

export default hierarchyFunctionMock;