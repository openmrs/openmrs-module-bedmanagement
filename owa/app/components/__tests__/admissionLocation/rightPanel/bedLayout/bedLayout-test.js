import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import BedLayout from 'components/admissionLocation/rightPanel/bedLayout';
import admissionLocationFunctions from 'components/__mocks__/admissionLocationFunctions-mock';

require('babel-polyfill');
require('components/__mocks__/location-mock');
const testProps = {
    admissionLocationFunctions: admissionLocationFunctions,
    sleep: (milliSec) => {
        return new Promise((resolve, reject) => {
            setTimeout(() => {
                return resolve(true);
            }, milliSec);
        });
    }
};

describe('BedLayout', () => {
    beforeAll(() => {
        var mock = new MockAdapter(axios);
        const data = {
            ward: {
                uuid: 'bb0e512e-d225-11e4-9c67-080027b662ec',
                display: 'VIP-100',
                name: 'VIP-100',
                description: 'VIP room 100'
            },
            bedLocationMappings: [
                {
                    rowNumber: 1,
                    columnNumber: 1,
                    bedNumber: 'vip-100',
                    bedId: 52,
                    bedUuid: '30dfc421-48ef-427f-9a67-001d2f75b8c5',
                    status: 'AVAILABLE',
                    bedType: {
                        id: 1,
                        name: 'deluxe',
                        displayName: 'deluxe bed',
                        description: 'DLXdfdfds'
                    }
                },
                {
                    rowNumber: 1,
                    columnNumber: 2,
                    bedNumber: null,
                    bedId: null,
                    bedUuid: null,
                    status: null,
                    bedType: null
                },
                {
                    rowNumber: 2,
                    columnNumber: 1,
                    bedNumber: null,
                    bedId: null,
                    bedUuid: null,
                    status: null,
                    bedType: null
                }
            ]
        };

        mock
            .onGet('https://192.168.33.10/openmrs/ws/rest/v1/admissionLocation/bb0e512e-d225-11e4-9c67-080027b662ec')
            .reply(200, data);
    });

    it('Should render bed block properly', async () => {
        const bedLayout = shallow(
            <BedLayout
                activeUuid="bb0e512e-d225-11e4-9c67-080027b662ec"
                admissionLocationFunctions={testProps.admissionLocationFunctions}
            />
        );
        await testProps.sleep(100);
        bedLayout.update();

        expect(bedLayout.find('BedLayoutRow').length).toBe(2);
        expect(bedLayout.state()).toEqual({
            bedlayouts: [
                undefined,
                [
                    undefined,
                    {
                        bedId: 52,
                        bedNumber: 'vip-100',
                        bedType: {
                            description: 'DLXdfdfds',
                            displayName: 'deluxe bed',
                            id: 1,
                            name: 'deluxe'
                        },
                        bedUuid: '30dfc421-48ef-427f-9a67-001d2f75b8c5',
                        columnNumber: 1,
                        rowNumber: 1,
                        status: 'AVAILABLE'
                    },
                    {
                        bedId: null,
                        bedNumber: null,
                        bedType: null,
                        bedUuid: null,
                        columnNumber: 2,
                        rowNumber: 1,
                        status: null
                    }
                ],
                [
                    undefined,
                    {
                        bedId: null,
                        bedNumber: null,
                        bedType: null,
                        bedUuid: null,
                        columnNumber: 1,
                        rowNumber: 2,
                        status: null
                    }
                ]
            ],
            layoutColumn: 2,
            layoutRow: 2,
            loadingData: false
        });
        expect(shallowToJson(bedLayout)).toMatchSnapshot();

        const notYetSetBedLayout = shallow(
            <BedLayout
                activeUuid="baf83667-d225-11e4-as58-080027b662ec"
                admissionLocationFunctions={testProps.admissionLocationFunctions}
            />
        );
        await testProps.sleep(100);
        notYetSetBedLayout.update();

        expect(shallowToJson(notYetSetBedLayout)).toMatchSnapshot();
    });

    it('Should trigger event handler', async () => {
        const sypAddWardClickHandler = jest.spyOn(BedLayout.prototype, 'addWardClickHandler');
        const sypSetBedLayoutClickHandler = jest.spyOn(BedLayout.prototype, 'setBedLayoutClickHandler');
        const spyOnSetState = jest.spyOn(testProps.admissionLocationFunctions, 'setState');
        const notYetSetBedLayout = mount(
            <BedLayout
                activeUuid="baf83667-d225-11e4-as58-080027b662ec"
                admissionLocationFunctions={testProps.admissionLocationFunctions}
            />
        );
        await testProps.sleep(100);
        notYetSetBedLayout.update();

        notYetSetBedLayout
            .find('label')
            .at(0)
            .simulate('click');
        expect(sypAddWardClickHandler).toHaveBeenCalled();
        expect(spyOnSetState).toHaveBeenCalledWith({
            activePage: 'addEditLocation',
            pageData: {
                operation: 'add'
            },
            activeUuid: 'baf83667-d225-11e4-as58-080027b662ec'
        });

        notYetSetBedLayout
            .find('label')
            .at(1)
            .simulate('click');
        expect(sypSetBedLayoutClickHandler).toHaveBeenCalled();
        expect(spyOnSetState).toHaveBeenCalledWith({
            activePage: 'set-layout',
            pageData: {},
            activeUuid: 'baf83667-d225-11e4-as58-080027b662ec'
        });
    });
});
