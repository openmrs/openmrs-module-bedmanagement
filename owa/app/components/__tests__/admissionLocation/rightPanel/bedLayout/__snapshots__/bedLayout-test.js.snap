// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BedLayout Should render bed block properly 1`] = `
<div
  className="bed-layout"
>
  <BedLayoutRow
    admissionLocationFunctions={
      Object {
        "getActiveLocationUuid": [Function],
        "getAdmissionLocationByUuid": [Function],
        "getAdmissionLocations": [Function],
        "getBedTypes": [Function],
        "getChildAdmissionLocations": [Function],
        "getParentAdmissionLocation": [Function],
        "getState": [Function],
        "getVisitLocations": [Function],
        "notify": [Function],
        "reFetchAllAdmissionLocations": [Function],
        "setActiveLocationUuid": [Function],
        "setState": [Function],
      }
    }
    key="1"
    layoutColumn={2}
    layoutRow={2}
    loadAdmissionLocationLayout={[Function]}
    row={1}
    rowBeds={
      Array [
        undefined,
        Object {
          "bedId": 52,
          "bedNumber": "vip-100",
          "bedType": Object {
            "description": "DLXdfdfds",
            "displayName": "deluxe bed",
            "id": 1,
            "name": "deluxe",
          },
          "bedUuid": "30dfc421-48ef-427f-9a67-001d2f75b8c5",
          "columnNumber": 1,
          "rowNumber": 1,
          "status": "AVAILABLE",
        },
        Object {
          "bedId": null,
          "bedNumber": null,
          "bedType": null,
          "bedUuid": null,
          "columnNumber": 2,
          "rowNumber": 1,
          "status": null,
        },
      ]
    }
  />
  <BedLayoutRow
    admissionLocationFunctions={
      Object {
        "getActiveLocationUuid": [Function],
        "getAdmissionLocationByUuid": [Function],
        "getAdmissionLocations": [Function],
        "getBedTypes": [Function],
        "getChildAdmissionLocations": [Function],
        "getParentAdmissionLocation": [Function],
        "getState": [Function],
        "getVisitLocations": [Function],
        "notify": [Function],
        "reFetchAllAdmissionLocations": [Function],
        "setActiveLocationUuid": [Function],
        "setState": [Function],
      }
    }
    key="2"
    layoutColumn={2}
    layoutRow={2}
    loadAdmissionLocationLayout={[Function]}
    row={2}
    rowBeds={
      Array [
        undefined,
        Object {
          "bedId": null,
          "bedNumber": null,
          "bedType": null,
          "bedUuid": null,
          "columnNumber": 1,
          "rowNumber": 2,
          "status": null,
        },
      ]
    }
  />
</div>
`;

exports[`BedLayout Should render bed block properly 2`] = `
<div
  className="location option"
>
  <label
    className="btn btn-primary"
    onClick={[Function]}
  >
    Add Child Admission Location
  </label>
  <label
    className="btn btn-primary"
    onClick={[Function]}
  >
    Set Bed Layout
  </label>
</div>
`;
